tests/unit/accounts/test_views.py

```diff
@@ -1065,12 +1098,86 @@ def test_webauthn_validate(self, monkeypatch, pyramid_request, has_recovery_code
             [] if has_recovery_codes else [pretend.call(pyramid_request, user)]
         )
 
+        assert _remember_device.calls == (
+            []
+            if not remember_device
+            else [
+                pretend.call(pyramid_request, pyramid_request.response, 1, "webauthn")
+            ]
+        )
+
         assert result == {
             "success": "Successful WebAuthn assertion",
             "redirect_to": "foobar",
         }
 
 
+class TestRememberDevice:
+    def test_check_remember_device_token_valid(self):
+        token_service = pretend.stub(loads=lambda *a: {"user_id": str(1)})
+        request = pretend.stub(
+            cookies=pretend.stub(get=lambda *a, **kw: "token"),
+            find_service=lambda interface, **kwargs: {
+                ITokenService: token_service,
+            }[interface],
+        )
+        assert views._check_remember_device_token(request, 1)
+
+    def test_check_remember_device_token_invalid_no_cookie(self):
+        request = pretend.stub(
+            cookies=pretend.stub(get=lambda *a, **kw: ""),
+        )
+        assert not views._check_remember_device_token(request, 1)
+
+    def test_check_remember_device_token_invalid_bad_token(self):
+        token_service = pretend.stub(loads=pretend.raiser(TokenException))
+        request = pretend.stub(
+            cookies=pretend.stub(get=lambda *a, **kw: "token"),
+            find_service=lambda interface, **kwargs: {
+                ITokenService: token_service,
+            }[interface],
+        )
+        assert not views._check_remember_device_token(request, 1)
+
+    def test_check_remember_device_token_invalid_wrong_user(self):
+        token_service = pretend.stub(loads=lambda *a: {"user_id": str(999)})
+        request = pretend.stub(
+            cookies=pretend.stub(get=lambda *a, **kw: "token"),
+            find_service=lambda interface, **kwargs: {
+                ITokenService: token_service,
+            }[interface],
+        )
+        assert not views._check_remember_device_token(request, 1)
+
+    def test_remember_device(self, monkeypatch, pyramid_request):
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏looks like you're recreating a `pyramid_request` inline of the test function, so you don't need the fixture, and you're not using the `monkeypatch` fixture either - should they be removed?

If the only part of `pyramid_request` you want to override is the `find_service`, you could express like:

```python
pyramid_request.find_service = lambda interface, **kwargs: {
    ITokenService: token_service
}[interface]
```

And remove the other parts that aren't necessary if the test works without them.